src/clients/QBittorrent.ts

```diff
@@ -244,57 +259,78 @@ export default class QBittorrent implements TorrentClient {
 	 */
 	async generateCorrectSavePath(
 		searchee: Searchee,
-		{ save_path },
-	): Promise<{ save_path: string }> {
-		const subfolderContentLayout =
-			await this.isSubfolderContentLayout(searchee);
+		torrentInfo: TorrentInfo,
+	): Promise<string> {
+		const subfolderContentLayout = await this.isSubfolderContentLayout(
+			searchee,
+			torrentInfo,
+		);
 		if (subfolderContentLayout) {
-			return {
-				save_path: join(save_path, stripExtension(searchee.name)),
-			};
+			return dirname(torrentInfo.content_path);
 		}
-		return { save_path };
+		return torrentInfo.save_path;
+	}
+
+	/*
+	@return array of query results
+	 */
+	async getAllTorrentInfo(): Promise<TorrentInfo[]> {
+		const responseText = await this.request("/torrents/info", "");
+		return JSON.parse(responseText);
 	}
+
+	/*
+	@param hash the hash of the torrent or undefined for all torrents
+	@return array of query results
+	 */
+	async getTorrentInfo(hash: string): Promise<TorrentInfo[]> {
+		const torrents = await this.getAllTorrentInfo();
+		return torrents.filter(
+			(torrent) =>
+				hash === torrent.hash ||
+				hash === torrent.infohash_v1 ||
+				hash === torrent.infohash_v2,
+		);
+	}
+
 	/*
 	@param searchee the searchee we are querying about
 	@return object with save_path, autoTMM, isComplete, and category from qBit
 	 */
 	async getTorrentConfiguration(
-		searchee: Searchee,
+		searchee: SearcheeWithInfoHash,
 	): Promise<TorrentConfiguration> {
-		const responseText = await this.request(
-			"/torrents/info",
-			`hashes=${searchee.infoHash}`,
-			X_WWW_FORM_URLENCODED,
-		);
-		const searchResult = JSON.parse(responseText).find(
-			(e) => e.hash === searchee.infoHash,
-		) as TorrentInfo;
-		if (searchResult === undefined) {
+		const searchResult = await this.getTorrentInfo(searchee.infoHash);
+		if (searchResult.length === 0) {
 			throw new Error(
 				"Failed to retrieve data dir; torrent not found in client",
 			);
 		}
 
-		const { progress, save_path, auto_tmm, category } = searchResult;
+		const { save_path, state, auto_tmm, category } = searchResult[0];
+		const isComplete = [
+			"uploading",
+			"pausedUP",
+			"queuedUP",
+			"stalledUP",
+			"checkingUP",
+			"forcedUP",
```

Comment:
Proactively cover v5?

In qBittorrent 5.0, pausedUP and pausedDL have been renamed to stoppedUP and stoppedDL.

Reply:
I didn't know that. Will fix.